Add overflow ellipsis and tooltips to output title-bar

hriday-panchasara opened the pull request:
fixes #537

avoids title-bar texts being mixed and makes diagrams more readable

Signed-off-by: hriday-panchasara <hriday.panchasara@ericsson.com>

3 output views opened and resized to their min-height:
![Screen Shot 2021-11-02 at 1 47 05 PM](https://user-images.githubusercontent.com/92893187/139931609-c1b0c799-4bbb-4be4-b11d-0dff791ade7a.png)

IbrahimFradj reviewed (approved):
LGTM, thank you. Maybe just a small remark in your commit i think we do it:

1.  Title 
2.  Description
3.  Fixes 
4.  signed-off

bhufmann commented:
While I agree that the issue of overflowing of the title needs to be fixed, I think having a minimum size will remove the possibility to have a graph with smaller height. For example, it's interesting to have a small CPU usage line chart on top of other views. 

![image](https://user-images.githubusercontent.com/413424/140559269-1b3296ed-4963-4aef-90e9-1e4d659facf4.png)

What do think about the alternative to add eclipse (e.g. "CPU U...") to the title and don't overflow the title box and also have tooltip that will pop-up with full title name if the user hovers over the title? With this, you can make the minimum height much smaller. Please let me know.

hriday-panchasara commented:
Removed output view minimum height bound, and implemented ellipsis and tooltip to deal with title bar text overflow:

![Screen Shot 2021-11-05 at 4 35 35 PM](https://user-images.githubusercontent.com/92893187/140581817-4afe1dea-6814-4a30-91f6-989b2c9ccf3c.png)

bhufmann reviewed (approved):
Thanks. Looks good to me.

mirsky-work commented:
> Removed output view minimum height bound, and implemented ellipsis and tooltip to deal with title bar text overflow:

Looks good. Don't forget to update the PR title and description and also clean the commits history.

hriday-panchasara commented:
> > Removed output view minimum height bound, and implemented ellipsis and tooltip to deal with title bar text overflow:
> 
> Looks good. Don't forget to update the PR title and description and also clean the commits history.

Updated the PR title and amended the commit message. Let me know if any other changes are needed.

IbrahimFradj commented:
> Updated the PR title and amended the commit message. Let me know if any other changes are needed.

I think the changes should also be in the commit message. 
Slogan to shorten to <= 50 characters.

bhufmann reviewed (changes requested):
End result looks good, but could you squash both commits? The one where you limit the height is not needed. Also, make sure that the commit message is formatted (as Ibrahim suggested)

hriday-panchasara commented:
> End result looks good, but could you squash both commits? The one where you limit the height is not needed. Also, make sure that the commit message is formatted (as Ibrahim suggested)

Hey Bernd, I've squashed the commits and formatted the commit message as requested.

bhufmann reviewed (approved):
Looks good to me. Thanks for the update!
